ethereum-consensus/src/error.rs

```diff
@@ -207,6 +211,28 @@ pub enum InvalidBlsToExecutionChange {
     PublicKeyMismatch(BlsPublicKey),
 }
 
+#[derive(Debug, Error)]
+pub enum InvalidConsolidation {
+    #[error("validator with index {0} is not in state")]
+    InvalidIndex(ValidatorIndex),
+    #[error("pending consolidations queue full")]
+    PendingConsolidationsQueueFull,
+    #[error("insufficient consolidation churn limit")]
+    InsufficientConsolidationChurnLimit,
+    #[error("source validator index equals target validator index {0}")]
+    EqualSourceAndTargetIndices(ValidatorIndex),
+    #[error("validator {0} is not active in the current epoch {1}")]
+    InactiveValidator(ValidatorIndex, Epoch),
+    #[error("exit initiated for validator {0}")]
+    ExitInitiatedForValidator(ValidatorIndex),
+    #[error("consolidation in epoch {consolidation_epoch} is not eligible for processing in current epoch {current_epoch}")]
+    EarlyConsolidation { current_epoch: Epoch, consolidation_epoch: Epoch },
+    #[error("validator {0} missing execution layer withdrawal credentials")]
+    MissingExecutionWithdrawalCredentials(ValidatorIndex),
+    #[error("source withdrawal credentials {source_address} do not equal target withdrawal credentials {target_address}")]
+    WithdrawalCredentialsMismatch { source_address: ByteVector<32>, target_address: ByteVector<32> },
```

Comment:
```suggestion
    WithdrawalCredentialsMismatch { source_address: crate::ExecutionAddress, target_address: crate::ExecutionAddress },
```

imo this is a more precise (and so more helpful) error, I would expect it complicates assembling the error but would consider it -- don't think its blocking merge tho

Reply:
yea, I was considering this but opted for the full credential. I can make the change.